src/zope/interface/adapter.py

```diff
@@ -430,6 +430,26 @@ def subscribe(self, required, provided, value):
 
         self.changed(self)
 
+    def subscribed(self, required, provided, subscriber):
```

Comment:
This method is almost identical to `registered`, except for which list it operates on (`_adapters` or `_subscribers`), the hardcoded name, and the last line. 

Now that I see that, it's probably worth refactoring to a single implementation.